client/pages/post.jsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      caption: '',
+      location: '',
+      postType: 'memory',
+      userId: 1,
+      file: null
+    };
+    this.fileInputRef = React.createRef();
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleCaptionChange = this.handleCaptionChange.bind(this);
+    this.handleLocationChange = this.handleLocationChange.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+
+  }
+
+  handleCaptionChange(event) {
+    this.setState({ caption: event.target.value });
+  }
+
+  handleLocationChange(event) {
+    this.setState({ location: event.target.value });
+  }
+
+  handleChange(event) {
+    this.setState({
+      file: URL.createObjectURL(event.target.files[0])
+    });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    const formData = new FormData();
+    formData.append('caption', this.state.caption);
+    formData.append('location', this.state.location);
+    formData.append('postType', this.state.postType);
+    formData.append('userId', this.state.userId);
+    formData.append('image', this.fileInputRef.current.files[0]);
+    fetch('/api/posts', {
+      method: 'POST',
+      body: formData
+    })
+      .then(response => response.json())
+      .then(() => {
+        this.setState({
+          caption: '',
+          location: ''
+        });
+        this.fileInputRef.current.value = null;
+      })
+      .catch(err => {
+        throw err;
+      });
+  }
+
+  render() {
+    return (
+      <div className="container">
+        <form className="memory-form" onSubmit={this.handleSubmit}>
+          <div className="row">
+              <div className="image-upload row">
```

Comment:
Do you really need a row inside a row here? See if you can simplify your markup and delete some unneeded css properties.